core/Test/Tasty/Ingredients/ConsoleReporter.hs

```diff
@@ -613,28 +616,33 @@ stringWidth = length
 
 -- (Potentially) colorful output
 ok, fail, skipped, infoOk, infoFail :: (?colors :: Bool) => String -> IO ()
-fail     = output BoldIntensity   Vivid Red
-ok       = output NormalIntensity Dull  Green
-skipped  = output NormalIntensity Dull  Magenta
-infoOk   = output NormalIntensity Dull  White
-infoFail = output NormalIntensity Dull  Red
+fail     = output failFormat
+ok       = output okFormat
+skipped  = output skippedFormat
+infoOk   = output infoOkFormat
+infoFail = output infoFailFormat
 
 output
   :: (?colors :: Bool)
-  => ConsoleIntensity
-  -> ColorIntensity
-  -> Color
+  => ConsoleFormat
   -> String
   -> IO ()
-output bold intensity color str
+output format = withConsoleFormat format . putStr
+
+withConsoleFormat
+  :: (?colors :: Bool)
```

Comment:
What does `?colors` signify? I don't think I've seen this syntax before.

Reply:
This is an implicit parameter, already present in the code. In this case it gets "bound by call graph".

See https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#implicit-parameters